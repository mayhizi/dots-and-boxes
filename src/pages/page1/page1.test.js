import React from "react"
import { shallow, mount } from "enzyme"

import Page1Component from "./page1.component"

describe("<Page1Component />", () => {
  it("renders correctly", () => {
    const tree = shallow(<Page1Component />)
    expect(tree).toMatchSnapshot()
  })
  it("it should contain an input", () => {
    const tree = shallow(<Page1Component />)
    // const instance = tree.getInstance()
    // expect(getByTestId(tree.container, "input")).toContain("input")
    expect(tree.find("input")).toHaveLength(1)
  })
  it("it should contain a button", () => {
    const tree = shallow(<Page1Component />)
    // const instance = tree.getInstance()
    // expect(getByTestId(tree.container, "input")).toContain("input")
    console.log(tree.find("button"))
    expect(tree.find("button")).toHaveLength(1)
  })
  it("it should show an error text when button clicked and input is empty", () => {
    const tree = mount(<Page1Component />)
    console.log("text", tree.text())
    tree.find("input").simulate("change", { target: { value: "" } })

    tree.find("button").simulate("click")
    console.log("text", tree.text())

    expect(tree.text()).toContain("not empty")
  })
})
