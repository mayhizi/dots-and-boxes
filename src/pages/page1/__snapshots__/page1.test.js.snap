// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page1Component /> renders correctly 1`] = `
<div>
  <input
    data-testid="input"
    type="text"
  />
  <div />
  <button
    onClick={[Function]}
  >
    submit
  </button>
</div>
`;
